services/web-server/src/loaders/WorkerManager.js

```diff
@@ -0,0 +1,53 @@
+import DataLoader from 'dataloader';
+import sift from 'sift';
+
+export default ({ workerManager }) => {
+  // const WMWorkerTypeSummaries = new DataLoader(queries =>
+  //   Promise.all(
+  //     queries.map(async (filter) => {
+  //       const summaries = await workerManager.listWorkerTypes();
+  //
+  //       return filter ? sift(filter, summaries) : summaries;
+  //     })
+  //   )
+  // );
+
+  const WMWorkerTypeSummaries = new DataLoader(queries => {
+    return Promise.all(
+      queries.map( ({ filter }) => {
+        const summaries = [
```

Comment:
Let's put mock data in a separate directory. Maybe we can name the directory `fixtures`. It should cleanup easier in the future.

Reply:
I am not sure if this change will actually make the future cleanup easier. I will have to make changes in two (or maybe more than two) places instead of one - I'll have to change an import somewhere, and remove the fixture from the dir

Reply:
The goal is to make sure this file doesn't differ much whether it's using mock or real data. I think the change in this file will be minimal (1 line deleted and another uncommented). The only other thing would be to delete the fixture file.

```js
import DataLoader from 'dataloader';
import sift from 'sift';
// the only line to delete once we are ready to use real data
import summaries from '../fixtures/workerTypeSummaries';

export default ({ workerManager }) => {
  const WorkerManagerWorkerTypeSummaries = new DataLoader(queries =>
    Promise.all(
      queries.map(async (filter) => {
        // const summaries = await workerManager.listWorkerTypes();
  
        return filter ? sift(filter, summaries) : summaries;
      })
    )
  );

  return {
    WorkerManagerWorkerTypeSummaries,
  };
};
```

Another thing is that as long as there's content inside the `fixtures` directory, we'll know that there's still some cleanup work to be done.

Reply:
That's exactly what I am talking about: instead of deleting the fixture data and uncommenting the actual function, I'll have to delete the import, change the function and also also delete the external file. The cleanup will be harder if it's in a separate dir. 

Why shouldn't this file differ much whether it's using mock data or not?

Reply:
> Why shouldn't this file differ much whether it's using mock data or not?

It's easier to detect errors when there are less changes in the loader file (smaller diff in business logic) once we stop using the mock data. For example, it took me a while to notice that the mapping function in the commented code uses `async` whereas the mock data uses a normal function and then use `Promise.resolve` when returning the result.

Reply:
I'm not sure how would a separate file help detect errors. For all I know, it would take you even longer to notice anything because they are not side by side - personally, it's much more difficult to compare things when they are not side by side (and also it's harder to write functions that return mock data too)

> mapping function in the commented code uses async whereas the mock data uses a normal function and then use Promise.resolve when returning the result.

You mean that an error? How? That function has to be asynchronous. So I use `Promise.resolve` so that it returns a `Promise`

Reply:
> I'm not sure how would a separate file help detect errors

Having mock data located outside the loader file will create a smaller diff to the business logic. A smaller diff in the business logic (the loader file) makes detecting errors easier.

> personally, it's much more difficult to compare things when they are not side by side

We won't need to compare the mock data. It is meant to be completely removed once we are able to query real data.

> it's harder to write functions that return mock data too

It doesn't need to be a function. `src/fixtures/workerTypeSummaries.js` could be as simple as

```jsx
module.exports = [
  {
    name: 'banana',
    pendingCapacity: 0,
    runningCapacity: 1,
    pendingTasks: 0,
    lastActive: new Date(),
    lastResolved: new Date(),
    failedNumber: 0,
    exceptionNumber: 1,
    unscheduledNumber: 2,
    provider: 'aws',
  },
  {
    name: 'pineapple',
    pendingCapacity: 1,
    runningCapacity: 3,
    pendingTasks: 1,
    lastActive: new Date('December 17, 1995 03:24:00'),
    lastResolved: new Date('December 17, 1995 03:24:00'),
    failedNumber: 1,
    exceptionNumber: 2,
    unscheduledNumber: 3,
    provider: 'gcp',
  },
];
```

Reply:
> A smaller diff in the business logic (the loader file) makes detecting errors easier.
I don't see how.

> It is meant to be completely removed once we are able to query real data
I believe it wasn't me who brought that point up. I know that this data is going to be removed

> It doesn't need to be a function
It may be so. I used this function and this mock data for my convenience, to make things faster, in the way I saw convenient for me. This is not going to stay. My point is, for me, it's more convenient to use this in this particular way. It's not going to stay here for much longer - I am hoping to finish the ui Pretty Soon(TM) and move on to the back end to make it return actual data. It will be easier for me to clean this up and work with it if stays where it is now

Reply:
Sounds good. We can keep the mock data in the same file.